blueocean-dashboard/src/main/js/components/Branches.jsx

```diff
@@ -58,6 +62,7 @@ export default class Branches extends Component {
         }
         const cleanBranchName = decodeURIComponent(branch.name);
         const runDetailsUrl = buildRunDetailsUrl(branch.organization, pipeline.fullName, cleanBranchName, latestRun.id, 'pipeline');
+        const historyButtonUrl = `${buildPipelineUrl(branch.organization, pipeline.fullName)}/activity?branch=${branch.name}`;
```

Comment:
As noted in the previous comment, I think this needs to be:
`${buildPipelineUrl(branch.organization, pipeline.fullName)}/activity?branch=${encodeURIComponent(branch.name)}`;